presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/PackageAPI.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teradata.prestomanager.agent.api;
+
+import com.teradata.prestomanager.agent.PrestoCommand;
+import com.teradata.prestomanager.agent.PrestoInstaller;
+import com.teradata.prestomanager.agent.PrestoUninstaller;
+import com.teradata.prestomanager.agent.PrestoUpgrader;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+import javax.inject.Singleton;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import static com.google.common.io.Files.getFileExtension;
+import static javax.ws.rs.client.Entity.text;
+import static javax.ws.rs.core.Response.Status.ACCEPTED;
+import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
+
+@Path("/package")
+@Api(description = "the package API")
+@Singleton
+public final class PackageAPI
+{
+    private static PackageType installedPackageType;
+
+    @PUT
+    @Consumes({"text/plain"})
+    @Produces({"text/plain"})
+    @ApiOperation(value = "Install presto using rpm or tarball")
+    @ApiResponses(value = {
+            @ApiResponse(code = 202, message = "Acknowledged request"),
+            @ApiResponse(code = 400, message = "Bad package format or Invalid url")
+    })
+    public synchronized Response install(@NotNull String locationToFetchPackage,
```

Comment:
I do not believe that we should have `@NotNull` annotation here. It doesn't hurt, but it doesn't help significantly either, and we don't use it in any of the other API classes.

We shouldn't be calling this method, so there's little risk that the lack of documentation will cause problems.
If it's a functional constraint, it should be enforced in the program itself, via `requireNonNull`.